### PR TITLE
Fix Vue builds

### PR DESCRIPTION
**Pull request by qmonmert**
All Vue builds failed currently, with this error:

`ERROR org.springframework.boot.SpringApplication - Application run failed
java.lang.IllegalStateException: No subdirectories found for mandatory directory location 'file:./config/*/'.`

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
Upstream bug: https://github.com/spring-projects/spring-boot/issues/26627

**Comment by qmonmert** [comment]
> spring-projects/spring-boot#26627

@mshima yes I saw that

**Comment by mshima** [comment]
Just for reference.

**Comment by qmonmert** [comment]
@mshima I try

**Comment by qmonmert** [comment]
@mshima it works :)

**Comment by DanielFran** [comment]
@qmonmert @mshima Better to revert the upgrade to spring-boot 2.4.6?

**Comment by qmonmert** [comment]
@DanielFran it works with Angular

**Comment by mshima** [comment]
@qmonmert try renaming webpack/index.js to env.js.
Probably require('webpack') is conflicting.

**Review by mshima — APPROVED** [review]
Just missing cleanup at https://github.com/jhipster/generator-jhipster/blob/main/generators/cleanup.js

File: generators/cleanup.js [hunk 1]
```diff
@@ -304,6 +304,13 @@ function cleanupOldFiles(generator) {
       generator.removeFile('.npmrc');
     }
   }
+  if (generator.isJhipsterVersionLessThan('7.0.2') && generator.jhipsterConfig) {
+    if (generator.jhipsterConfig.clientFramework === VUE) {
+      generator.removeFile('config/index.js');
```
Comment (DanielFran) [inline generators/cleanup.js:309]:
```suggestion
      generator.removeFile('${VUE_DIR}config/index.js');
```

File: generators/cleanup.js [hunk 2]
```diff
@@ -304,6 +304,13 @@ function cleanupOldFiles(generator) {
...
+      generator.removeFile('config/dev.env.js');
```
Comment (DanielFran) [inline generators/cleanup.js:310]:
```suggestion
      generator.removeFile('${VUE_DIR}config/dev.env.js');
```

File: generators/cleanup.js [hunk 3]
```diff
@@ -304,6 +304,13 @@ function cleanupOldFiles(generator) {
...
+      generator.removeFile('config/prod.env.js');
```
Comment (DanielFran) [inline generators/cleanup.js:311]:
```suggestion
      generator.removeFile('${VUE_DIR}config/prod.env.js');
```

**Review by DanielFran — CHANGES_REQUESTED** [review]
@qmonmert I believe the path is not correct for the files you are removing

File: generators/cleanup.js (hunk 1 above)
Comment (qmonmert) [reply generators/cleanup.js:309]: ok :)

File: generators/cleanup.js (hunk 2 above)
Comment (qmonmert) [reply generators/cleanup.js:310]: ok :)

File: generators/cleanup.js (hunk 3 above)
Comment (qmonmert) [reply generators/cleanup.js:311]: ok :)

File: generators/cleanup.js (hunk 1 above)
Comment (mshima) [reply generators/cleanup.js]: I think it was correct

File: generators/cleanup.js (hunk 2 above)
Comment (mshima) [reply generators/cleanup.js:310]: It's correct

**Review by DanielFran — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
Adding a bounty on this, for the time to investigate and fix the builds
Plz claim it, @qmonmert

**Comment by qmonmert** [comment]
@pascalgrimaud https://opencollective.com/generator-jhipster/expenses/41624 :)

**Comment by pascalgrimaud** [comment]
@qmonmert : approved

**Pull request merged**